### ISSUE TITLE
TimeoutError: _DoCommand

### ISSUE THREAD

**Issue by ghost** [kevinvincent/ha-wyzesense#29]
Trying to get this to work on my Raspberrypi 3B+, but getting Timeout Error: _DoCommand during startup. The dongle status light is blue, so I believed it is has been recognized, but still getting timeout. Currently running Hassio v0.98.5.
====================================
2019-09-11 21:35:43 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant.
2019-09-11 21:35:44 INFO (MainThread) [homeassistant.setup] Setup of domain logger took 0.0 seconds.
2019-09-11 21:35:45 INFO (MainThread) [homeassistant.setup] Setting up http
2019-09-11 21:35:45 INFO (MainThread) [homeassistant.setup] Setup of domain http took 0.0 seconds.
2019-09-11 21:35:45 INFO (MainThread) [homeassistant.setup] Setting up system_log
2019-09-11 21:35:45 INFO (MainThread) [homeassistant.setup] Setup of domain system_log took 0.0 seconds.
2019-09-11 21:35:45 INFO (MainThread) [homeassistant.setup] Setting up recorder
2019-09-11 21:35:47 WARNING (Recorder) [homeassistant.components.recorder] Ended unfinished session (id=9 from 2019-09-12 01:20:00.915227)
2019-09-11 21:35:48 INFO (MainThread) [homeassistant.setup] Setup of domain recorder took 2.7 seconds.
2019-09-11 21:35:48 INFO (MainThread) [homeassistant.setup] Setting up updater
2019-09-11 21:35:48 INFO (MainThread) [homeassistant.setup] Setup of domain updater took 0.0 seconds.
2019-09-11 21:35:50 INFO (MainThread) [homeassistant.setup] Setting up ssdp
2019-09-11 21:35:50 INFO (MainThread) [homeassistant.setup] Setup of domain ssdp took 0.0 seconds.
2019-09-11 21:35:51 INFO (SyncWorker_19) [homeassistant.util.package] Attempting install of aiofiles==0.4.0
2019-09-11 21:35:51 INFO (MainThread) [homeassistant.setup] Setting up sun
2019-09-11 21:35:51 INFO (MainThread) [homeassistant.setup] Setup of domain sun took 0.2 seconds.
2019-09-11 21:35:52 INFO (MainThread) [homeassistant.setup] Setting up person
2019-09-11 21:35:52 INFO (MainThread) [homeassistant.setup] Setting up lovelace
2019-09-11 21:35:52 INFO (MainThread) [homeassistant.setup] Setup of domain lovelace took 0.0 seconds.
2019-09-11 21:35:52 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for wyzesense which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant.
2019-09-11 21:35:52 INFO (MainThread) [homeassistant.setup] Setting up group
2019-09-11 21:35:52 INFO (MainThread) [homeassistant.setup] Setup of domain group took 0.0 seconds.
2019-09-11 21:35:54 INFO (MainThread) [homeassistant.setup] Setting up api
2019-09-11 21:35:54 INFO (MainThread) [homeassistant.setup] Setting up system_health
2019-09-11 21:35:54 INFO (MainThread) [homeassistant.setup] Setup of domain system_health took 0.0 seconds.
2019-09-11 21:35:54 INFO (MainThread) [homeassistant.setup] Setting up auth
2019-09-11 21:35:54 INFO (MainThread) [homeassistant.setup] Setup of domain auth took 0.0 seconds.
2019-09-11 21:35:54 INFO (MainThread) [homeassistant.setup] Setting up webhook
2019-09-11 21:35:54 INFO (MainThread) [homeassistant.setup] Setup of domain webhook took 0.0 seconds.
2019-09-11 21:35:55 INFO (MainThread) [homeassistant.setup] Setting up script
2019-09-11 21:35:55 INFO (MainThread) [homeassistant.setup] Setup of domain script took 0.0 seconds.
2019-09-11 21:35:55 INFO (MainThread) [homeassistant.setup] Setting up config
2019-09-11 21:35:55 INFO (MainThread) [homeassistant.setup] Setting up websocket_api
2019-09-11 21:35:55 INFO (MainThread) [homeassistant.setup] Setup of domain websocket_api took 0.0 seconds.
2019-09-11 21:35:55 INFO (MainThread) [homeassistant.setup] Setting up history
2019-09-11 21:35:55 INFO (MainThread) [homeassistant.setup] Setup of domain history took 0.0 seconds.
2019-09-11 21:35:55 INFO (MainThread) [homeassistant.setup] Setup of domain api took 0.4 seconds.
2019-09-11 21:35:55 INFO (MainThread) [homeassistant.setup] Setting up onboarding
2019-09-11 21:35:55 INFO (MainThread) [homeassistant.setup] Setting up automation
2019-09-11 21:35:55 INFO (MainThread) [homeassistant.setup] Setup of domain automation took 0.0 seconds.
2019-09-11 21:35:55 INFO (MainThread) [homeassistant.setup] Setup of domain config took 0.2 seconds.
2019-09-11 21:35:55 INFO (MainThread) [homeassistant.setup] Setup of domain onboarding took 0.2 seconds.
2019-09-11 21:35:55 INFO (MainThread) [homeassistant.setup] Setup of domain person took 3.4 seconds.
2019-09-11 21:37:09 INFO (SyncWorker_14) [homeassistant.util.package] Attempting install of aiogithubapi==0.2.0
2019-09-11 21:38:43 INFO (SyncWorker_5) [homeassistant.util.package] Attempting install of integrationhelper==0.2.2
2019-09-11 21:40:16 INFO (MainThread) [homeassistant.setup] Setting up hacs
2019-09-11 21:40:16 INFO (MainThread) [hacs] 
-------------------------------------------------------------------
HACS (Home Assistant Community Store)

Version: 0.13.2
This is a custom integration
If you have any issues with this you need to open an issue here:
https://github.com/custom-components/hacs/issues
-------------------------------------------------------------------

2019-09-11 21:40:17 INFO (MainThread) [homeassistant.setup] Setting up met
2019-09-11 21:40:17 INFO (MainThread) [homeassistant.setup] Setup of domain met took 0.0 seconds.
2019-09-11 21:40:17 INFO (SyncWorker_13) [homeassistant.util.package] Attempting install of wyzesense==0.0.4
2019-09-11 21:40:17 INFO (SyncWorker_6) [homeassistant.loader] Loaded weather from homeassistant.components.weather
2019-09-11 21:40:17 INFO (MainThread) [homeassistant.setup] Setting up weather
2019-09-11 21:40:17 INFO (MainThread) [homeassistant.setup] Setup of domain weather took 0.0 seconds.
2019-09-11 21:40:17 INFO (MainThread) [homeassistant.components.weather] Setting up weather.met
2019-09-11 21:40:22 INFO (MainThread) [hacs.repository.integration.custom-components.hacs] Registration complete
2019-09-11 21:40:22 INFO (MainThread) [hacs.data] Restore started
2019-09-11 21:40:22 INFO (MainThread) [hacs.data] Restore done
2019-09-11 21:40:22 INFO (MainThread) [homeassistant.setup] Setup of domain hacs took 5.8 seconds.
2019-09-11 21:40:22 INFO (SyncWorker_16) [homeassistant.loader] Loaded sensor from homeassistant.components.sensor
2019-09-11 21:40:22 INFO (MainThread) [homeassistant.setup] Setting up sensor
2019-09-11 21:40:22 INFO (MainThread) [homeassistant.setup] Setup of domain sensor took 0.0 seconds.
2019-09-11 21:40:22 INFO (MainThread) [homeassistant.components.sensor] Setting up sensor.hacs
2019-09-11 21:41:56 INFO (SyncWorker_10) [homeassistant.util.package] Attempting install of retry==0.9.2
2019-09-11 21:43:07 INFO (MainThread) [homeassistant.setup] Setting up binary_sensor
2019-09-11 21:43:07 INFO (SyncWorker_9) [homeassistant.loader] Loaded google_translate from homeassistant.components.google_translate
2019-09-11 21:43:09 INFO (MainThread) [homeassistant.setup] Setting up mobile_app
2019-09-11 21:43:10 INFO (MainThread) [homeassistant.setup] Setting up cloud
2019-09-11 21:43:13 INFO (MainThread) [homeassistant.setup] Setting up zeroconf
2019-09-11 21:43:13 INFO (MainThread) [homeassistant.setup] Setting up frontend
2019-09-11 21:43:13 INFO (MainThread) [homeassistant.setup] Setup of domain frontend took 0.1 seconds.
2019-09-11 21:43:13 INFO (MainThread) [homeassistant.components.binary_sensor] Setting up binary_sensor.wyzesense
2019-09-11 21:43:13 INFO (MainThread) [homeassistant.setup] Setup of domain mobile_app took 4.1 seconds.
2019-09-11 21:43:13 DEBUG (SyncWorker_5) [custom_components.wyzesense.binary_sensor] WYZESENSE v0.0.4
2019-09-11 21:43:13 DEBUG (SyncWorker_5) [custom_components.wyzesense.binary_sensor] Attempting to open connection to hub at /dev/hidraw0
2019-09-11 21:43:13 DEBUG (SyncWorker_5) [custom_components.wyzesense.wyzesense_custom] Start Inquiry...
2019-09-11 21:43:13 DEBUG (SyncWorker_5) [custom_components.wyzesense.wyzesense_custom] ===> Sending: Packet: Cmd=4327, Payload=<None>
2019-09-11 21:43:13 DEBUG (SyncWorker_5) [custom_components.wyzesense.wyzesense_custom] Sending: b'aa55430327016c'
2019-09-11 21:43:13 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Trying to parse: b'55aa43042801016f'
2019-09-11 21:43:13 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Received: b'55aa43042801016f'
2019-09-11 21:43:13 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] <=== Received: Packet: Cmd=4328, Payload=b'01'
2019-09-11 21:43:14 DEBUG (SyncWorker_5) [custom_components.wyzesense.wyzesense_custom] Inquiry returns 1
2019-09-11 21:43:14 DEBUG (SyncWorker_5) [custom_components.wyzesense.wyzesense_custom] ===> Sending: Packet: Cmd=5314, Payload=b'ff'
2019-09-11 21:43:14 DEBUG (SyncWorker_5) [custom_components.wyzesense.wyzesense_custom] Sending: b'aa55530414ff0269'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Trying to parse: b'55aa5314ff026555aa530e3500000000000000000314ff02ab55aa530315016a'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Received: b'55aa5314ff0265'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] <=== Received: Packet: Cmd=53FF, Payload=ACK(5314)
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Trying to parse: b'55aa530e3500000000000000000314ff02ab55aa530315016a55aa530315016a'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Received: b'55aa530e3500000000000000000314ff02ab'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] <=== Received: Packet: Cmd=5335, Payload=b'00000000000000000314ff'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] ===> Sending: Packet: Cmd=53FF, Payload=ACK(5335)
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Sending: b'aa555335ff0286'
2019-09-11 21:43:14 INFO (Thread-2) [custom_components.wyzesense.wyzesense_custom] LOG: time=1969-12-31T19:00:00, data=b'14ff'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Trying to parse: b'55aa530315016a55aa530315016a'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Received: b'55aa530315016a'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] <=== Received: Packet: Cmd=5315, Payload=<None>
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] ===> Sending: Packet: Cmd=53FF, Payload=ACK(5315)
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Sending: b'aa555315ff0266'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Trying to parse: b'55aa530315016a'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Received: b'55aa530315016a'
2019-09-11 21:43:14 DEBUG (SyncWorker_5) [custom_components.wyzesense.wyzesense_custom] Start GetSensors...
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] <=== Received: Packet: Cmd=5315, Payload=<None>
2019-09-11 21:43:14 DEBUG (SyncWorker_5) [custom_components.wyzesense.wyzesense_custom] ===> Sending: Packet: Cmd=532E, Payload=<None>
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] ===> Sending: Packet: Cmd=53FF, Payload=ACK(5315)
2019-09-11 21:43:14 DEBUG (SyncWorker_5) [custom_components.wyzesense.wyzesense_custom] Sending: b'aa5553032e0183'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Sending: b'aa555315ff0266'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Trying to parse: b'55aa5303320187'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Received: b'55aa5303320187'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] <=== Received: Packet: Cmd=5332, Payload=<None>
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] ===> Sending: Packet: Cmd=53FF, Payload=ACK(5332)
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Sending: b'aa555332ff0283'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] ===> Sending: Packet: Cmd=5333, Payload=b'0000016d2323892f'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Sending: b'aa55530b330000016d2323892f02fc'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Trying to parse: b'55aa532eff027f55aa53042f020187'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Received: b'55aa532eff027f'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] <=== Received: Packet: Cmd=53FF, Payload=ACK(532E)
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Trying to parse: b'55aa53042f02018755aa5303320187'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Received: b'55aa53042f020187'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] <=== Received: Packet: Cmd=532F, Payload=b'02'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] ===> Sending: Packet: Cmd=53FF, Payload=ACK(532F)
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Sending: b'aa55532fff0280'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Trying to parse: b'55aa530332018755aa5333ff0284'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Received: b'55aa5303320187'
2019-09-11 21:43:14 DEBUG (SyncWorker_5) [custom_components.wyzesense.wyzesense_custom] 2 sensors reported, waiting for each one to report...
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] <=== Received: Packet: Cmd=5332, Payload=<None>
2019-09-11 21:43:14 DEBUG (SyncWorker_5) [custom_components.wyzesense.wyzesense_custom] ===> Sending: Packet: Cmd=5330, Payload=b'02'
2019-09-11 21:43:14 INFO (MainThread) [homeassistant.setup] Setting up tts
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] ===> Sending: Packet: Cmd=53FF, Payload=ACK(5332)
2019-09-11 21:43:14 DEBUG (SyncWorker_5) [custom_components.wyzesense.wyzesense_custom] Sending: b'aa55530430020188'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Sending: b'aa555332ff0283'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] ===> Sending: Packet: Cmd=5333, Payload=b'0000016d232389a4'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Sending: b'aa55530b330000016d232389a40371'
2019-09-11 21:43:14 INFO (MainThread) [homeassistant.setup] Setting up map
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Trying to parse: b'55aa5333ff0284'
2019-09-11 21:43:14 INFO (MainThread) [homeassistant.setup] Setup of domain map took 0.0 seconds.
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Received: b'55aa5333ff0284'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] <=== Received: Packet: Cmd=53FF, Payload=ACK(5333)
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Trying to parse: b'55aa5330ff028155aa530b3137373837383139450352'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Received: b'55aa5330ff0281'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] <=== Received: Packet: Cmd=53FF, Payload=ACK(5330)
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Trying to parse: b'55aa530b3137373837383139450352'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Received: b'55aa530b3137373837383139450352'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] <=== Received: Packet: Cmd=5331, Payload=b'3737383738313945'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] ===> Sending: Packet: Cmd=53FF, Payload=ACK(5331)
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Sending: b'aa555331ff0282'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Sensor 1/2, MAC:7787819E
2019-09-11 21:43:14 INFO (MainThread) [homeassistant.setup] Setting up logbook
2019-09-11 21:43:14 INFO (MainThread) [homeassistant.setup] Setup of domain logbook took 0.0 seconds.
2019-09-11 21:43:14 INFO (MainThread) [homeassistant.setup] Setting up panel_custom
2019-09-11 21:43:14 INFO (MainThread) [homeassistant.setup] Setup of domain panel_custom took 0.0 seconds.
2019-09-11 21:43:14 INFO (MainThread) [homeassistant.setup] Setup of domain zeroconf took 1.6 seconds.
2019-09-11 21:43:14 INFO (SyncWorker_8) [homeassistant.loader] Loaded notify from homeassistant.components.notify
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Trying to parse: b'55aa5333ff0284'
2019-09-11 21:43:14 INFO (SyncWorker_6) [homeassistant.loader] Loaded esphome from homeassistant.components.esphome
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] Received: b'55aa5333ff0284'
2019-09-11 21:43:14 DEBUG (Thread-2) [custom_components.wyzesense.wyzesense_custom] <=== Received: Packet: Cmd=53FF, Payload=ACK(5333)
2019-09-11 21:43:14 INFO (MainThread) [homeassistant.setup] Setting up hassio
2019-09-11 21:43:14 INFO (MainThread) [homeassistant.setup] Setting up notify
2019-09-11 21:43:14 INFO (MainThread) [homeassistant.setup] Setup of domain notify took 0.0 seconds.
2019-09-11 21:43:16 INFO (MainThread) [homeassistant.components.notify] Setting up notify.mobile_app
2019-09-11 21:43:16 INFO (MainThread) [homeassistant.setup] Setup of domain cloud took 6.7 seconds.
2019-09-11 21:43:16 INFO (MainThread) [homeassistant.setup] Setup of domain tts took 2.2 seconds.
2019-09-11 21:43:18 INFO (MainThread) [homeassistant.setup] Setting up default_config
2019-09-11 21:43:18 INFO (SyncWorker_16) [homeassistant.loader] Loaded stream from homeassistant.components.stream
2019-09-11 21:43:18 INFO (SyncWorker_12) [homeassistant.loader] Loaded homekit_controller from homeassistant.components.homekit_controller
2019-09-11 21:43:18 INFO (MainThread) [homeassistant.setup] Setting up stream
2019-09-11 21:43:18 INFO (MainThread) [homeassistant.setup] Setup of domain stream took 0.0 seconds.
2019-09-11 21:43:18 INFO (MainThread) [homeassistant.setup] Setup of domain default_config took 0.2 seconds.
2019-09-11 21:43:19 INFO (MainThread) [homeassistant.setup] Setup of domain hassio took 4.7 seconds.
2019-09-11 21:43:23 WARNING (MainThread) [homeassistant.components.binary_sensor] Setup of platform wyzesense is taking over 10 seconds.
2019-09-11 21:43:24 ERROR (MainThread) [homeassistant.components.binary_sensor] Error while setting up platform wyzesense
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 150, in _async_setup_platform
    await asyncio.wait_for(asyncio.shield(task), SLOW_SETUP_MAX_WAIT)
  File "/usr/local/lib/python3.7/asyncio/tasks.py", line 442, in wait_for
    return fut.result()
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/wyzesense/binary_sensor.py", line 85, in setup_platform
    result = ws.List()
  File "/config/custom_components/wyzesense/wyzesense_custom.py", line 507, in List
    sensors = self._GetSensors()
  File "/config/custom_components/wyzesense/wyzesense_custom.py", line 479, in _GetSensors
    self._DoCommand(Packet.GetSensorList(count), cmd_handler, timeout=self._CMD_TIMEOUT * count)
  File "/config/custom_components/wyzesense/wyzesense_custom.py", line 384, in _DoCommand
    raise TimeoutError("_DoCommand")
TimeoutError: _DoCommand
2019-09-11 21:43:24 INFO (MainThread) [homeassistant.setup] Setup of domain binary_sensor took 17.4 seconds.
2019-09-11 21:43:24 INFO (MainThread) [homeassistant.components.binary_sensor] Setting up binary_sensor.updater
2019-09-11 21:43:24 INFO (MainThread) [homeassistant.bootstrap] Home Assistant initialized in 462.37s
2019-09-11 21:43:24 INFO (MainThread) [homeassistant.core] Starting Home Assistant
2019-09-11 21:43:24 INFO (MainThread) [hacs] Loading known repositories
2019-09-11 21:43:24 INFO (MainThread) [homeassistant.core] Timer:starting
2019-09-11 21:48:20 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/aiohttp/web_protocol.py", line 275, in data_received
    messages, upgraded, tail = self._request_parser.feed_data(data)
  File "aiohttp\_http_parser.pyx", line 523, in aiohttp._http_parser.HttpParser.feed_data
aiohttp.http_exceptions.BadStatusLine: invalid HTTP method

**Comment by foley803** [comment]
Same here, raspi 3 with Hassio installation. Service is missing after a reboot and only one of the 3 sensors is still available.

Error in log:
```
2019-09-13 16:54:10 ERROR (MainThread) [homeassistant.components.binary_sensor] Error while setting up platform wyzesense
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 150, in _async_setup_platform
    await asyncio.wait_for(asyncio.shield(task), SLOW_SETUP_MAX_WAIT)
  File "/usr/local/lib/python3.7/asyncio/tasks.py", line 442, in wait_for
    return fut.result()
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/wyzesense/binary_sensor.py", line 85, in setup_platform
    result = ws.List()
  File "/config/custom_components/wyzesense/wyzesense_custom.py", line 507, in List
    sensors = self._GetSensors()
  File "/config/custom_components/wyzesense/wyzesense_custom.py", line 479, in _GetSensors
    self._DoCommand(Packet.GetSensorList(count), cmd_handler, timeout=self._CMD_TIMEOUT * count)
  File "/config/custom_components/wyzesense/wyzesense_custom.py", line 384, in _DoCommand
    raise TimeoutError("_DoCommand")
TimeoutError: _DoCommand
```

Full log: [https://pastebin.com/cYgiHCyR](url)

**Comment by photinus** [comment]
Seeing the same issue, identical logs as the last few users to report it. Running HASSOS through ESXI. Tried rebooting, tried hot removing and re-adding the usb receiver to the vm, no dice on any attempts.

**Comment by atinsley** [comment]
I've found a temporary workaround for adding new sensors since these errors cause the wyzesense.scan service to not be exposed in Home Assistant.  It's clunky, but it works for now.  Sensors are actually paired with the bridge hardware, not anything in software, so we can pair them to the bridge using another virtual machine.

1. Create a new VM running Ubuntu Server
2. Attach the Wyze Sense bridge to the VM
3. Install [WyzeSensePy](https://github.com/HclX/WyzeSensePy/blob/master/sample.py) by running "git clone https://github.com/HclX/WyzeSensePy.git"
4. Run "dmesg | grep hidraw" to get the device ID for the Wyze Sense bridge.
5. Run "cd WyzeSensePy/" to move into the WyzeSensePy directory
6. Run "pip3 install docopt" to install the docopt package.  You may need to install the pip3 package if this is a new Ubuntu Server VM.
7 Run "sudo python3 --device /dev/hidraw1" in the Terminal.  Replace /dev/hidraw1 with the device ID you obtained in step 4.
8. Hit P to enter pairing mode.
9. Use the pin tool to pair the sensor to the bridge.
10. When you're done pairing all your sensors, reconnect the bridge to your Home Assistant PC, Raspi, or virtual machine.  Make sure the light is blue.
11. Sensors will appear in Home Assistant the first time they're triggered

Hope this helps somebody else get around this issue until it's fixed.  I've added 5 new sensors since I first ran into this "TimeoutError: _DoCommand" error.

**Comment by photinus** [comment]
Interestingly enough, I think one of my sensors was potentially causing the issue. After I pulled the battery from one of my door contact sensors the error went away after a restart. It was blinking the red led a few times on a regular basis compared to the others that would not blink unless opened/closed. I think it was in some error state.

**Comment by foley803** [comment]
I had a sensor blinking extra blinks too. It turns out the battery was dying. @photinus 's solution worked for me too, then I replaced the battery and everything went back to normal (the service loaded without errors after a reboot and I was able to re-add the sensors. I do have an extra phantom sensor without a MAC address still in the unused entities page of HA, but it's not listed in the entity registry. I wonder if the sensor starts returning garbage data when the battery gets low. I'll try to investigate on another machine using WyzeSensePy and my dead battery.

**Comment by photinus** [comment]
So, I think one of my sensors freaking out put me in a bad state. After troubleshooting, the "_DoCommand" error is coming directly out of the attempt to get the list of sensors. I get the same error if I run the hclx scripts to manually try and work with the dongle. Adjusting the timeout doesn't fix it. I think what has happened is one of my sensors got registered with the dongle with a bad mac or something and now I cant seem to remove it. I tried unpairing all my sensors and the hclx library still shows I have one device paired, but when attempting to list it I run into that same error.

With that in mind, my thought of a fix would be to not attempt to pull in the list of sensors from the dongle. Have the add-on maintain it's own registry in a json file or something of the registered sensors. They would be added to the internal registry when the device first sends in a status change or at the time of pairing. Similar to the current state and would accomplish the same result. Thoughts?

**Comment by photinus** [comment]
I managed to find a totally hackish way around the issue for me. Replace your wyzesense_custom.py with this attachment.

What I was seeing, connecting the dongle to another system and using the hclx python scripts, it would connect to the dongle and see events just fine. I could Pair and Un-pair sensors without issue, but if I tried to list the sensors from the dongle, it would report 1 more sensor then I had paired and would never return the full list causing the timeout. I just hacked mine to reduce the count returned from the dongle by one to prevent the timeout error waiting for the phantom sensor.

[wyzesense_custom.zip](https://github.com/kevinvincent/ha-wyzesense/files/3636863/wyzesense_custom.zip)

**Comment by photinus** [comment]
Possibly a better fix for those looking for one, give this a try:
https://github.com/photinus/ha-wyzesense

It removes the dongle scan and stores the sensors in a local text file. First time it's run, you'll likely see sensors disappear but they should automatically be re-created when they check in for the first time.

**Comment by Madseason41** [comment]
> 
> 
> I managed to find a totally hackish way around the issue for me. Replace your wyzesense_custom.py with this attachment.
> 
> What I was seeing, connecting the dongle to another system and using the hclx python scripts, it would connect to the dongle and see events just fine. I could Pair and Un-pair sensors without issue, but if I tried to list the sensors from the dongle, it would report 1 more sensor then I had paired and would never return the full list causing the timeout. I just hacked mine to reduce the count returned from the dongle by one to prevent the timeout error waiting for the phantom sensor.
> 
> [wyzesense_custom.zip](https://github.com/kevinvincent/ha-wyzesense/files/3636863/wyzesense_custom.zip)

After finding a sensor that was blinking red and going through the process of "wyze.remove" and "wyze.scan" to add a new one, I experienced the "DoCommand" error. Nothing I did seemed to resolve the issue. I replaced the file with the first one from this thread, and everything returned to normal. Thanks!

**Comment by AK5nowman** [comment]
Appears I'm having the same issue. Here is my info if its useful. 

When I restart home assistant I see a log entry of "_DoCommand, retrying in 1 seconds..." over and over before home assistant proceeds with the startup without wyze. When this happens my CPU is pegged out at 100 pct usage and stays that way for the duration of home assistants run time.

System:
Fresh Docker.
Hass version 0.100.2
ha-wyzesense version 0.0.4 - Manual installation

Logs: https://hastebin.com/ajigahebap.sql

When I just spun up a version 0.92.0 of HASS it appears to startup with no issue. I'd prefer to be able to stay current on HASS release, however.

Please let me know what other information would be of use and I can post it. Thank you!

**Comment by jamman9000** [comment]
> Possibly a better fix for those looking for one, give this a try:
> https://github.com/photinus/ha-wyzesense
> 
> It removes the dongle scan and stores the sensors in a local text file. First time it's run, you'll likely see sensors disappear but they should automatically be re-created when they check in for the first time.

Thank you! I added the repository, uninstalled the github/kevinvincent version and installed your edited version. Restarted HA and the error has disappeared. Much cleaner startup and now I can see/use the services.

From what I read this version may fix the last issue frustrating me where all my wyze entities disappear on a HA restart... only to return upon change of state via transmission of the data from the sensor activating. Before, it was hard to get rid of old entities no longer in use and delete out of HA, now these are hard to keep in. lol

**Comment by Mincka** [comment]
Hello,
Exact same issues than @jamman9000. My sensors were unavailable at each restart before a state change. I can confirm than the fix from @photinus works perfectly, thank you. It's sad that the pull request was not merge with the one from the autodetection. I had to manually merge the changes from @photinus in the latest version. 

But now it works great with my Raspberry Pi 4. This integration is however still an "unofficial hack" and I feel more confident using the Zigbee sensors with my Conbee II.

**Comment by photinus** [comment]
I still need to test the auto detection pull request. Kevin added me as a
maintainer so I can help push these forward. My hope is to test that pull
request with my own to make for the next major release.

**Comment by xtalker** [comment]
I see this is closed, but I'm still having problems here.  I had the issue exactly as described above.

For some reason the WyzeSense addon (under Community) no longer shows up?   Not sure if this is related or not so I updated /config/custom_components/wyzesense manually from github and rebooted. There are no longer any errors in the log, but I have one contact sensor that just won't show.  I tried to remove and it reported that this sensor was not found so then added it successfully, but it still doesn't show up in the state table.  All other sensors seem to work ok?

HA ver: 0.103.4

WyzeSense files:
`-rwxr--r--    1 root     root          7457 Dec  6 13:35 binary_sensor.py
-rwxr--r--    1 root     root           281 Dec  6 13:35 manifest.json
-rwxr--r--    1 root     root           258 Dec  6 13:35 services.yaml
-rwxr--r--    1 root     root         17966 Dec  6 13:35 wyzesense_custom.py`

Any help appreciated!
